enterprise/backend/src/metabase_enterprise/workspaces/copying.clj

```diff
@@ -0,0 +1,144 @@
+                                              (merge
+                                               (-> (select-keys transform t2-transform-mirrored-keys)
+                                                   ;; this should be happening elsewhere, not on write, for now ok
+                                                   (update :name str "_DUP"))
+                                               {:target {:type "table"
+                                                         :database (:id database)
+                                                         :schema mirror-schema-name
+                                                         :name mirror-table-name}}))]
+    (assoc entities-info :transforms [(assoc-in transform [:mirror :transform] mirror)])))
+
+(comment
+  ;; 
+  (t2/select-one :model/Transform)
+  ;;=> (toucan2.instance/instance
+  ;;    :model/Transform
+  ;;    {:description "x",
+  ;;     :dependency_analysis_version 2,
+  ;;     :name "overwrite_existing_test",
+  ;;     :source
+  ;;     {:type :query,
+  ;;      :query
+  ;;      {:lib/type :mbql/query,
+  ;;       :lib/metadata
+  ;;       (metabase.lib.metadata.invocation-tracker/invocation-tracker-provider
+  ;;        (metabase.lib.metadata.cached-provider/cached-metadata-provider
+  ;;         (metabase.lib-be.metadata.jvm/->UncachedApplicationDatabaseMetadataProvider 2))),
+  ;;       :database 2,
+  ;;       :stages
+  ;;       [{:lib/type :mbql.stage/mbql,
+  ;;         :aggregation
+  ;;         [[:sum
+  ;;           #:lib{:uuid "ed6c85c7-9d2f-48e5-899a-ca77a56791f0"}
+  ;;           [:field
+  ;;            {:effective-type :type/Float, :lib/uuid "3e975bcc-4444-41d8-94ad-f107ec83419f", :base-type :type/Float}
+  ;;            259]]],
+  ;;         :source-table 32,
+  ;;         :breakout
+  ;;         [[:field
+  ;;           {:effective-type :type/DateTimeWithLocalTZ,
+  ;;            :base-type :type/DateTimeWithLocalTZ,
+  ;;            :lib/uuid "32cf8766-5a19-430d-a006-4482cd8b9493",
+  ;;            :temporal-unit :month}
+  ;;           261]]}]}},
+  ;;     :source_type :mbql,
+  ;;     :creator_id 1,
+  ;;     :updated_at #t "2025-11-18T10:07:10.024448Z",
+  ;;     :id 1,
+  ;;     :entity_id "otpI4D8HlmpmnIaesJzAL",
+  ;;     :target {:type "table", :database 2, :name "tabi_tabi", :schema "public"},
+  ;;     :created_at #t "2025-11-18T10:03:57.366345Z"})
+  )
+
+;; TODO: proper docstring
+(defn mirror-entities!
+  "Create duplicated namespace specific entities.
+
+  Input:
+  TBD, when setteled the schema will be added
+  For now it takes tables and
+
+  bla bla
+
+  RETURNING
+  some map with mappings of old entities to new entities"
+  [workspace ; some map
+   {:keys [_check-outs
+           inputs
+           _tables
+           _transforms
+           _nodes]
+    :as entities-info}]
+  ;; - ignoring multiple entites from multiple databases atm -- hence following "get database" is ok
+  ;; - inputs are tables that should be made RO later
```

Comment:
If we restrict this at the top level (in the api shape and workspace table schema) we can skip scattering assumption comments around the place.